packages/server/src/agents.storage.ts

```diff
@@ -0,0 +1,410 @@
+            return;
+          }
+          logger.error('Error creating table:', error);
+        });
+      }
+      logger.debug('Database tables created successfully');
+    } catch (error) {
+      logger.error('Error during agents_controller initialisation:', error);
+      throw error;
+    }
+  }
+
+  private async init_models_config() {
+    try {
+      logger.debug('Initializing models configuration');
+      const q = new Postgres.Query(
+        `SELECT EXISTS(SELECT 1 FROM models_config WHERE id = 1)`
+      );
+      const result = await Postgres.query<{ exists: boolean }>(q);
+      if (!result[0].exists) {
+        logger.debug('Models configuration not found, creating default config');
+
+        const fast: ModelLevelConfig = {
+          provider: ModelProviders.OpenAI,
+          model_name: 'gpt-4o-mini',
+          description: 'Optimized for speed and simple tasks.',
+        };
+        const smart: ModelLevelConfig = {
+          provider: ModelProviders.Anthropic,
+          model_name: 'claude-3-5-sonnet-latest',
+          description: 'Optimized for complex reasoning.',
+        };
+        const cheap: ModelLevelConfig = {
+          provider: ModelProviders.OpenAI,
+          model_name: 'gpt-4o-mini',
+          description: 'Good cost-performance balance.',
+        };
+        logger.debug('Models configuration not found, creating default config');
+        const q = new Postgres.Query(
+          `INSERT INTO models_config (id, fast, smart, cheap) VALUES (1, ROW($1, $2, $3), ROW($4, $5, $6), ROW($7, $8, $9))`,
+          [
+            fast.provider,
+            fast.model_name,
+            fast.description,
+            smart.provider,
+            smart.model_name,
+            smart.description,
+            cheap.provider,
+            cheap.model_name,
+            cheap.description,
+          ]
+        );
+        await Postgres.query(q);
+      } else {
+        logger.debug('Models configuration already exists, skipping creation.');
+      }
+    } catch (error) {
+      logger.error('Error during agents_controller initialisation:', error);
+      throw error;
+    }
+    1;
+  }
+
+  private async initialize() {
+    try {
+      if (this.initialized) {
+        return;
+      }
+      await Postgres.connect({
+        database: process.env.POSTGRES_DB as string,
+        host: process.env.POSTGRES_HOST as string,
+        user: process.env.POSTGRES_USER as string,
+        password: process.env.POSTGRES_PASSWORD as string,
+        port: parseInt(process.env.POSTGRES_PORT as string),
+      });
+
+      await this.initalize_database();
+      await this.init_models_config();
+      await this.init_agents_config();
+      this.initialized = true;
+    } catch (error) {
+      logger.error('Error during agents_controller initialisation:', error);
+      throw error;
+    }
+  }
+
+  private parseAgentConfig(config: any): ModelLevelConfig {
```

Comment:
parsing could be more robust, this one assumes config is always a string format, will break if it changes

can you use AgentConfig for config?